src/python/pants/backend/python/lint/pyupgrade/rules.py

```diff
@@ -38,55 +36,34 @@ class PyUpgradeRequest(FmtRequest):
     name = PyUpgrade.options_scope
 
 
-@dataclass(frozen=True)
-class PyUpgradeResult:
-    process_result: FallibleProcessResult
-    original_snapshot: Snapshot
-
 
 @rule(level=LogLevel.DEBUG)
-async def run_pyupgrade(request: PyUpgradeRequest, pyupgrade: PyUpgrade) -> PyUpgradeResult:
```

Comment:
PyUpgrade was written a little funny, and not like the others. So I refactored it to match our other tools